src/value.rs

```
@@ -291,6 +291,55 @@ impl Value {
         self.ty.as_ref() == ty
     }
 
+    /// Get the default value of the given type.
+    ///
+    /// The default value serializes to a string of zeroes.
+    ///
+    /// ## Construction
+    ///
+    /// - `default( 1 )` = `()`
+    /// - `default( A + B )` = `default(A)`
+    /// - `default( A × B )` = `default(A) × default(B)`
+    pub fn default_of_type(ty: &Final) -> Self {
```

In 8784adaa9fc42b7b53e076ec7cbb257b04ae8b92:

Can we call this `zero_value` or something like that? I think "default" misleadingly implies that the returned value is important.

---

I borrowed Rust's `Default::default` terminology, where the default integer values are also zeroes. However, Rust describes this trait as _"A trait for giving a type a useful default value."_ which implies that each implementation should think about which default value is the most useful. I, on the other hand, always return the value that serializes to zeroes, so I can see that the name "default" can be misleading.

`Value::zero` or `Value::all_zeroes` could be better.

---

I think that using 0 as the "default" is only a convention for numeric types. For containers, the "empty container" is the default value. This is already an overload of two things that mean something quite different.

If stdlib were designed by people who could do math, zero would be an associated constant on `Add` (and one an associated constant on `Mul`), but I was told "this is not Haskell" whenever I suggested that things be sane or consistent. (And also there were no associated constants back then, so zero/one would've needed to be *methods* and that would've been kinda ugly.) So instead we have a `+` operator with no identity which is routinely used for noncommutative operations and you just have to guess what anything means. Might as well be C++ with side-effecting shift operators..

Anyway, for non-numerics and non-containers there is no convention and it's nonobvious to users what the "default value" is supposed to be. So I think we should avoid the trait except in those two places.